pyo3-macros-backend/src/pymethod.rs

```diff
@@ -191,18 +190,18 @@ pub fn impl_py_method_def(
 }
 
 fn impl_py_method_def_new(cls: &syn::Type, spec: &FnSpec) -> Result<TokenStream> {
-    let wrapper_ident = syn::Ident::new("__wrap", Span::call_site());
+    let wrapper_ident = syn::Ident::new("__pymethod__new__", Span::call_site());
     let wrapper = spec.get_wrapper_function(&wrapper_ident, Some(cls))?;
-    Ok(quote! {
-        impl _pyo3::impl_::pyclass::PyClassNewImpl<#cls> for _pyo3::impl_::pyclass::PyClassImplCollector<#cls> {
-            fn new_impl(self) -> ::std::option::Option<_pyo3::ffi::newfunc> {
-                ::std::option::Option::Some({
-                    #wrapper
-                    #wrapper_ident
-                })
-            }
+    Ok(quote! {{
+        impl #cls {
+            #[doc(hidden)]
+            #wrapper
         }
```

Comment:
Realising that the `wrapper` could be an associated method of the struct is interesting for a couple of reasons:
- We have some rather ugly code to recursively `replace_self` (in method argument types and return values) - if all Pymethod wrappers were placed on the struct like this, we won't need it.
- In addition, because the struct is a shared namespace, we could use this safeguard against multiple Python definitions with the same name. (That's exactly why I did it here; so that `#[new]` can only be implemented once.)

I'll probably play around with this in a follow-up PR.